server/modules/arenaImport/service/arenaImport/metaImportJobs/surveyCreatorJob.js

```diff
@@ -45,8 +46,10 @@ export default class SurveyCreatorJob extends Job {
     const surveyId = Survey.getId(survey)
 
     const activities = await ArenaSurveyFileZip.getActivities(arenaSurveyFileZip)
-
-    await ActivityLogManager.insertManyBatch(this.user, surveyId, activities, this.tx)
+    const activitiesOrderById = activities.sort(
+      (activityA, activityB) => ActivityLog.getDateCreated(activityA) - ActivityLog.getDateCreated(activityB)
+    )
+    await ActivityLogManager.insertManyBatch(this.user, surveyId, activitiesOrderById, this.tx)
```

Comment:
wouldn't it be better to export them in order already ?

Reply:
Yep, Also we are reusing the same function that we use for the activitylog and in that case I think that system logs are not exported... so, yes I think that it is better add a new Repository function only to export the activity log